src/lib/geo_location_utils.js

```diff
@@ -22,16 +20,53 @@ var locationmodeToIdFinder = {
     'country names': countryNameToISO3
 };
 
-exports.locationToFeature = function(locationmode, location, features) {
+function countryNameToISO3(countryName) {
+    for(var i = 0; i < countryIds.length; i++) {
+        var iso3 = countryIds[i];
+        var regex = new RegExp(countryRegex[iso3]);
+
+        if(regex.test(countryName.trim().toLowerCase())) return iso3;
+    }
+
+    Lib.log('Unrecognized country name: ' + countryName + '.');
+
+    return false;
+}
+
+function locationToFeature(locationmode, location, features) {
     if(!location || typeof location !== 'string') return false;
 
-    var locationId = getLocationId(locationmode, location);
+    var locationId = locationmodeToIdFinder[locationmode](location);
+    var features2;
+    var f, i;
 
     if(locationId) {
-        for(var i = 0; i < features.length; i++) {
-            var feature = features[i];
+        if(locationmode === 'USA-states') {
+            // Filter out features south of the equator
+            //
+            // This is important as the Natural Earth files
+            // include state/provinces from USA, Canada, Australia and Brazil
+            // which have some overlay in their two-letter ids. For example,
+            // 'WA' is used for both Washington state and Western Australia.
+            // As subunits from USA and Canada never conflict, filtering out features
+            // south of the equator suffices to fix https://github.com/plotly/plotly.js/issues/3779
+            //
+            // A better fix would have us add a "governing unit" properties in subunit features
+            // in the `sane-topojson` package to avoid conflicts.
+            features2 = [];
+            for(i = 0; i < features.length; i++) {
+                f = features[i];
+                if(f.properties && f.properties.ct && f.properties.ct[1] > 0) {
```

Comment:
The maximum latitude of Brazil is above 5N. It may be better to use greater than 6 here.

Reply:
No conflicting _centroids_ go above the equator though.